include/common/core/freertos_task.hpp

```diff
@@ -73,4 +74,28 @@ class FreeRTOSTask {
     std::function<void()> starter{};
 };
 
+template <uint32_t StackDepth,
+          template <template <typename> typename> typename TaskObj>
+struct TaskStarter {
+    TaskStarter() : queue{}, task_entry{queue}, task{task_entry} {}
```

Comment:
You could go further and remove the queue from this class. Rather than this class creating the `TaskEntry` instance it can be given a reference.

Reply:
Something like this?
```c++
// freertos_task.hpp
template <size_t StackDepth, typename TaskObj>
struct TaskStarter {
    explicit TaskStarter(TaskObj& task_entry) : task_entry{task_entry}... {}
    ...
};
// e.g. pipettes/core/tasks.cpp
auto capacitive_queue = freertos_queue::FreeRTOSMessageQueue<sensors::tasks::TaskMessage>{};
auto capacitive_task = sensors::tasks::CapacitiveSensorTask(capacitive_queue);
auto capacitive_starter = freertos_task::TaskStarter<512>(capacitive_task);
```

If so, I'd really like to not do that - it means that there's now three lines of boilerplate per task in tasks.cpp rather than just one.

Reply:
That's fine.